feature-libs/customer-ticketing/components/customer-ticketing-dialog/customer-ticketing-dialog.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Directive, ElementRef, HostListener } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import {
+  CustomerTicketingConfig,
+  MAX_ENTRIES_FOR_ATTACHMENT,
+  MAX_INPUT_CHARACTERS,
+  MAX_INPUT_CHARACTERS_FOR_SUBJECT,
+  MAX_SIZE_FOR_ATTACHMENT,
+} from '@spartacus/customer-ticketing/root';
+import {
+  FilesFormValidators,
+  FocusConfig,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+
+@Directive()
+export abstract class CustomerTicketingDialogComponent {
+  iconTypes = ICON_TYPE;
+  form: FormGroup;
+  inputCharactersLimit: number = this.getInputCharactersLimit;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'button',
+    focusOnEscape: true,
+  };
+
+  get messagesCharacterLeft(): number {
+    return (
+      this.inputCharactersLimit - (this.form.get('message')?.value?.length || 0)
+    );
+  }
+
+  get allowedTypes(): string[] | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.allowedTypes;
+  }
+
+  get getInputCharactersLimit(): number {
+    return (
+      this.customerTicketingConfig.customerTicketing?.inputCharactersLimit ||
```

Comment:
would like you think using `nullish coalescing` is better than the or operator for this case?

Let's say the customer sets it at 0, do you want it to evaluate to the lefthand side or the right handside (default value)

reason being, the OR operator will evaluate 0 as `falsy`, whereas the nullish opeartor evaluates as truthy.

Reply:
Yes, let's use nullish coalescing.